src/backend/distributed/commands/common.c

```diff
@@ -283,3 +285,61 @@ PreprocessDropDistributedObjectStmt(Node *node, const char *queryString,
 
 	return NodeDDLTaskList(NON_COORDINATOR_NODES, commands);
 }
+
+
+/*
+ * DropTextSearchDictObjectAddress returns list of object addresses in the drop tsdict
+ * statement if all objects are valid; else, returns an empty list of addresses.
+ */
+List *
+DropTextSearchDictObjectAddress(Node *node, bool missing_ok)
+{
+	DropStmt *stmt = castNode(DropStmt, node);
+
+	List *objectAddresses = NIL;
+
+	List *objName = NIL;
+	foreach_ptr(objName, stmt->objects)
+	{
+		Oid tsdictOid = get_ts_dict_oid(objName, missing_ok);
+		if (!OidIsValid(tsdictOid))
+		{
+			return NIL;
+		}
+
+		ObjectAddress *objectAddress = palloc0(sizeof(ObjectAddress));
+		ObjectAddressSet(*objectAddress, TSDictionaryRelationId, tsdictOid);
+		objectAddresses = lappend(objectAddresses, objectAddress);
+	}
+
+	return objectAddresses;
+}
+
+
+/*
+ * DropTextSearchConfigObjectAddress returns list of object addresses in the drop tsconfig
+ * statement if all objects are valid; else, returns an empty list of addresses.
+ */
+List *
+DropTextSearchConfigObjectAddress(Node *node, bool missing_ok)
+{
+	DropStmt *stmt = castNode(DropStmt, node);
+
+	List *objectAddresses = NIL;
+
+	List *objName = NIL;
+	foreach_ptr(objName, stmt->objects)
+	{
+		Oid tsconfigOid = get_ts_config_oid(objName, missing_ok);
+		if (!OidIsValid(tsconfigOid))
+		{
+			return NIL;
```

Comment:
for many other objects, we use `get_object_address` such as `AlterCollationOwnerObjectAddress`.

And, it seems that, on those cases PG sets the objectId to invalid: https://github.com/postgres/postgres/blob/3e071b2cf584052b0799ebbc4d38ecfdb5bd701a/src/backend/catalog/objectaddress.c#L924-L928


However, here we return `NIL`.

Wouldn't that cause confusion? For some objects, we'd check for `NIL` list for others we'd need to iterate over the list and check for `objectId == InvalidOid`?

I it might be better that here we also add `objectId = InvalidOid` to the list instead of returning `NIL`?